src/hotspot/share/opto/divnode.cpp

```diff
@@ -862,21 +1098,58 @@ const Type* UDivINode::Value(PhaseGVN* phase) const {
     return TypeInt::ONE;
   }
 
-  // Either input is BOTTOM ==> the result is the local BOTTOM
-  const Type *bot = bottom_type();
-  if( (t1 == bot) || (t2 == bot) ||
-      (t1 == Type::BOTTOM) || (t2 == Type::BOTTOM) )
-    return bot;
+  // TODO: Improve Value inference of both signed and unsigned division
+  const TypeInt* i1 = t1->isa_int();
+  const TypeInt* i2 = t2->isa_int();
+  assert(i1 != nullptr && i2 != nullptr, "");
+  if (i1->is_con() && i2->is_con()) {
+    return TypeInt::make(juint(i1->get_con()) / juint(i2->get_con()));
+  }
 
   // Otherwise we give up all hope
   return TypeInt::INT;
 }
 
 //------------------------------Idealize---------------------------------------
-Node *UDivINode::Ideal(PhaseGVN *phase, bool can_reshape) {
+Node* UDivINode::Ideal(PhaseGVN* phase, bool can_reshape) {
   // Check for dead control input
   if (in(0) && remove_dead_region(phase, can_reshape))  return this;
-  return NULL;
+  // Don't bother trying to transform a dead node
+  if( in(0) && in(0)->is_top() )  return nullptr;
```

Comment:
Please use correct style in new code here and other places (even for one liner):
```
   if (cond) {
     body;
   }
```

Reply:
Thanks for your reviews, I have changed the code style in these places.